### PR TITLE
Various updates to preventing course version change

### PR DESCRIPTION
**Pull request by bethanyaconnor**
- Add missing checks to prevent course version change
- Add a test that an exception is thrown when trying to update a unit group
- Disable family name/version year selectors for courses/units that should not change their unit group

Family name and version year disabled on the unit and course edit pages:
Screen Shot 2021-06-30 at 3 36 24 PM
Screen Shot 2021-06-30 at 3 35 25 PM

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Comment by bethanyaconnor** [comment]
Actually, I'm going to change part of this. I'll ping again when it's ready.

**Comment by bethanyaconnor** [comment]
Thinking about this more, there is some more work I'd like to do here but I think this is a good first step, particularly disabling the family name and version year inputs. There is a bug when setting the family name to empty, which this PR does not fix, but I'm wondering whether we can just live with that until all scripts move into a unit group and the course version owns the unit group.

so tl;dr I think this PR is worth reviewing now

File: dashboard/test/controllers/courses_controller_test.rb [hunk 1]
```diff
@@ -438,6 +438,29 @@ class CoursesControllerTest < ActionController::TestCase
     refute_nil unit_group.course_version
   end
 
+  test "update: cannot course version for unit groups" do
```
Comment (davidsbailey) [inline dashboard/test/controllers/courses_controller_test.rb]: nit: "cannot change"?

**Review by davidsbailey — APPROVED** [review]
This seems like a good start!

**Pull request merged**